components/tools/OmeroJava/test/integration/DuplicationTest.java

```diff
@@ -0,0 +1,1112 @@
+     */
+    private static class PlaneInfoKey {
+
+        private final int z, t, c;
+
+        /**
+         * Construct a new key.
+         * @param planeInfo the {@link PlaneInfo} to which this key is to correspond
+         */
+        PlaneInfoKey(PlaneInfo planeInfo) {
+            this.z = planeInfo.getTheZ().getValue();
+            this.t = planeInfo.getTheT().getValue();
+            this.c = planeInfo.getTheC().getValue();
+        }
+
+        @Override
+        public boolean equals(Object object) {
+            if (this == object) {
+                return true;
+            } else if (object instanceof PlaneInfoKey) {
+                final PlaneInfoKey other = (PlaneInfoKey) object;
+                return this.z == other.z && this.t == other.t && this.c == other.c;
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(getClass(), z, t, c);
+        }
+    }
+
+    private final List<Long> testImages = Collections.synchronizedList(new ArrayList<Long>());
+
+    /**
+     * Clear the list of test images.
+     */
+    @BeforeClass
+    public void clearTestImages() {
+        testImages.clear();
+    }
+
+    /**
+     * Delete the test images then clear the list.
+     * @throws Exception unexpected
+     */
+    @AfterClass
+    public void deleteTestImages() throws Exception {
+        final Delete2 delete = Requests.delete("Image", testImages);
+        doChange(root, root.getSession(), delete, true);
+        clearTestImages();
+    }
+
+    /**
+     * Assert that the given instances have the same property values.
+     * (The properties to compare are based on the behavior of {@link ModelMockFactory#createImage(int, int, int, int, int)}.)
+     * @param original the original instance
+     * @param duplicate the duplicate instance that is expected to correspond to the original
+     */
+    private static void assertSameProperties(Image original, Image duplicate) {
+        Assert.assertEquals(duplicate.getDescription().getValue(), original.getDescription().getValue());
+        Assert.assertEquals(duplicate.getFormat().getValue(), original.getFormat().getValue());
+        Assert.assertEquals(duplicate.getName().getValue(), original.getName().getValue());
+        Assert.assertEquals(duplicate.getSeries().getValue(), original.getSeries().getValue());
+    }
+
+    /**
+     * Assert that the given instances have the same property values.
+     * (The properties to compare are based on the behavior of {@link ModelMockFactory#createImage(int, int, int, int, int)}.)
+     * @param original the original instance
+     * @param duplicate the duplicate instance that is expected to correspond to the original
+     */
+    private static void assertSameProperties(Pixels original, Pixels duplicate) {
+        Assert.assertEquals(duplicate.getPhysicalSizeX().getUnit(), original.getPhysicalSizeX().getUnit());
```

Comment:
That could actually be useful for de-duplication on import: e.g., not creating a duplicate of many tags because an image finally returned (via whatever means) to the group in which it was originally tagged.